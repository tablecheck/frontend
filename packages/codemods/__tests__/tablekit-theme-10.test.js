const path = require('path');

const fs = require('fs-extra');
const jscodeshift = require('jscodeshift');
const getCodeshiftParser = require('jscodeshift/src/getParser');

const codeshift = require('../scripts/tablekit-theme-10');

const testFilesDirectory = path.resolve('./testFiles');
const files = fs.readdirSync(testFilesDirectory);

function getCodeshiftResult(fileContent) {
  return codeshift(
    { source: fileContent },
    {
      jscodeshift: jscodeshift.withParser(getCodeshiftParser('tsx', {}))
    }
  );
}

describe('tablekit-theme-10', () => {
  test.each(files)('should convert %s', (fileName) => {
    expect(
      getCodeshiftResult(
        fs.readFileSync(path.join(testFilesDirectory, fileName), {
          encoding: 'utf8'
        })
      )
    ).toMatchSnapshot();
  });
});
