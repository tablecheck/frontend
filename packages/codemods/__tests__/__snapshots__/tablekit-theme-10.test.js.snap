// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tablekit-theme-10 should convert tablekitTheme10_1.tsx 1`] = `
"//TODO fix temp imports
import {
  COLORS as BASE,
  TYPOGRAPHY as TYPOGRAPHY__TEMP,
  DEPTH,
  Spacing as Spacing__TEMP,
  ZIndex,
  Size,
  FieldHeight,
  OtherType,
} from \\"@tablekit/theme\\";

import { CLASSIC_COLORS, COLORS } from '@tablekit/other';
const TYPOGRAPHY = 'something about type';

const Spacing = 'true';

function calculateMinimumHeight(props) {
  const { minimumRows, theme } = props;
  const col = theme.colors.blue;
  const padding = parseFloat(Spacing__TEMP.L3) * 2;
  const typographySettings = TYPOGRAPHY__TEMP[props.font] || TYPOGRAPHY__TEMP.L2;
  const lineHeight = parseFloat(typographySettings.lineHeight);
  const font = TYPOGRAPHY__TEMP.L1;
  return \`\${padding + lineHeight * minimumRows}px\`;
}

const size: Size = Size.XXLarge;
let size2: Size;
let size3: Size;
type sizeObj = {
  [key in Size]: string;
};
type sizeTypeA = Size[];

const theme = {
  backgroundActiveColor: (props: any) =>
    hexToRgba(props.theme.colors.primary2, 0.2)
};

export const EventPreview = styled.div\`
  border-style: dashed;
  border-width: 1px;
  border-color: \${({ theme }) => theme.colors.border};
  border-color: \${({ theme }) => theme.colors.border};
  border-color: \${props => props.theme.colors.border};
  border-color: \${({ theme }) => theme.colors.border};
  border-color: \${({ ...props }) => props.theme.colors.border};
  border-color: \${(props) => props.theme.colors.border};
  border-color: \${(props) => {
    const { color } = props;
    if (!color)
      return props.theme.colors.text || props.theme.colors.text;
    if (typeof color === 'function') return color(props);
    return color;
  }};
  border-color: \${({
  color,
  theme
}) => color || theme.colors.border};
  border-color: \${(props) => props.color || props.theme.colors.border};
  border-color: \${function test({
  color,
  theme
}) {
    return color || theme.colors.border;
  }};
  border-color: \${({ color, theme }) =>
    color || theme.colors.border};
  padding: \${Spacing__TEMP.L2};
  color: \${COLORS.GRAY.L6};
  background-color: \${BASE.GRAY.L6};
  margin: \${Spacing__TEMP.L2} 0;
  \${TYPOGRAPHY__TEMP.L1};
  \${TYPOGRAPHY__TEMP.L1};
  \${typography('L1', 'L2', 'L3')()};
  \${(props) => {
    return (TYPOGRAPHY__TEMP[props.level] || TYPOGRAPHY__TEMP.L1).fieldValue;
  }};
  \${TYPOGRAPHY.EX};

  z-index: \${ZIndex.Topnav};
  height: \${FieldHeight.Small};
\`;

const EP = <EventPreview>Test</EventPreview>;
"
`;

exports[`tablekit-theme-10 should convert tablekitTheme10_2.ts 1`] = `
"import { COLORS } from \\"@tablekit/theme\\";

const t1 = (props) => css\`
  color: \${props.theme.colors.secondary};
\`;

const t2 = (props) => {
  return css\`
  color: \${props.theme.colors.secondary};
\`;
};

function getColor(props) {
  let borderColor = 'blue';
  if (props.isInvalid) {
    borderColor = props.theme.colors.danger;
  }
  return borderColor;
}

export const alertClassicTheme = {
  [ALERT_APPEARANCES.INFO]: {
    iconColor: ({ theme }) => theme.colors.iconFill
  },
  actionBtnColor: ({ theme }) => theme.colors.primary,
  boxShadowColor: COLORS.GRAY_TRANSLUCENT.L1,
  closeButton: {
    color: ({ theme }) => theme.colors.iconFill
  }
};

export const LoadingIndicator = () => {
  const theme = useTheme();
  return <Spinner color={theme.colors.secondary} />;
};

const result = {
  colors: {
    danger: ({ theme }) => theme.colors.danger
  }
};

type PropsType1 = {
  color?: string | Function;
};

interface PropsType2 {
  color?: string | Function;
}
"
`;

exports[`tablekit-theme-10 should convert tablekitTypescript2020.tsx 1`] = `
"import Button, { BUTTON_APPEARANCES } from '@tablekit/button';

const PrimaryButton = <Button appearance={BUTTON_APPEARANCES.PRIMARY} />;
"
`;

exports[`tablekit-theme-10 should convert tablekitUtils3.ts 1`] = `
"import { Spacing } from \\"@tablekit/theme\\";
import { css as utilCss, variant, MediaQueryPropType } from '@tablekit/utils';
import { css as emotionCss } from '@emotion/react';
import { css } from '@emotion/react';

function someUtil() {
  return () => 12;
}

type t2 = MediaQueryPropType;

const staticStyles = css\`
  color: blue;
  margin: 12px;
  padding: \${Spacing.L1};
\`;

const staticStyles2 = emotionCss\`
  color: blue;
  margin: 12px;
  padding: \${Spacing.L1};
\`;

export const transition = utilCss\`
  transition: 120ms ease-in-out;
  \${(props) => {
    return '';
  }}
\`;

const interpolatedStyles = utilCss\`
  color: \${({ theme, ...props }) => theme.colors.primary};
  color: \${(props2) => props2.theme.colors.primary};
  color: \${() => (true ? '12px' : '0px')};
  background-color: \${someUtil()};
  \${variant({ prop: 'test', variants: {} })};
  \${variant({ prop: () => 'test', variants: {} })};
  \${variant({
    prop: () => ({
      'min-width: 1200px': 'test'
    }),
    variants: {}
  })};\${variant({
  prop: {
    'min-width: 1200px': 'test'
  },
  variants: {}
})};
  margin: 12px;
  padding: \${Spacing.L1};
\`;

const testC = ({ theme, ...props }) => {
  const styles = utilCss\`
    color: \${({ theme }) => theme.colors.primary};
  \`({ theme, ...props });
  const test = theme.emotionCss.other;
  return styles;
};

function test({ theme, ...props }) {
  const styles = utilCss\`
    color: \${({ theme }) => theme.colors.primary};
  \`({ theme, ...props });
  const test = theme.emotionCss.other;
  return styles;
}

function test2(props) {
  const styles = utilCss\`
      color: \${({ theme }) => theme.colors.primary};
      color: \${({ isTrue, theme }) => (isTrue ? theme.colors.primary : 'blue')};
      color: \${({ red, blue, green }) => \`rgb(\${red}, \${blue}, \${green})\`};
      color: \${({ red, blue, green }) =>
        'rgb(' + red + ', ' + blue + ', ' + green + ')'};
      margin: \${({ spacing, isSmall, other }) => {
        if (isSmall) return 0;
        switch (spacing) {
          case 'small':
            return 6;
          case 'regular':
            return 12;
        }
        return 18;
      }};
    \`(props);
  const test = props.theme.emotionCss.other;
  return styles;
}

function test3({ isTrue, theme, red, blue, green, spacing, isSmall }) {
  const styles = utilCss\`
    color: \${({ isTrue, theme }) => (isTrue ? theme.colors.primary : 'blue')};
    color: \${({ red, blue, green }) => \`rgb(\${red}, \${blue}, \${green})\`};
    margin: \${({ spacing, isSmall }) => {
      if (isSmall) return 0;
      switch (spacing) {
        case 'small':
          return 6;
        case 'regular':
          return 12;
      }
      return 18;
    }};
  \`({ isTrue, theme, red, blue, green, spacing, isSmall });
  const test = theme.emotionCss.other;
  return styles;
}

function test4({ theme, ...props }) {
  const styles = utilCss\`
    color: \${({ theme }) => theme.colors.primary};
  \`({ theme });
  const test = props.theme.emotionCss.other;
  return styles;
}

const test5 = ({ spacingMax, isSmall, ...props }) =>
  utilCss\`
    margin: \${({ spacing, isSmall, other }) => {
      if (isSmall) return 0;
      switch (spacing) {
        case 'small':
          return 6;
        case 'regular':
          return 12;
      }
      return 18;
    }};
  \`({ spacing: spacingMax, isSmall, ...props });
"
`;

exports[`tablekit-theme-10 should convert typescript2020.tsx 1`] = `
"import React, {
  Node,
  Node as OtherNode,
  Fragment,
  Fragment as Frag,
  Component,
  StatelessFunctionalComponent,
  useState
} from 'react';
import React2 from 'react';
import { GenericNode } from 'some-import';
import { GenericNode2 } from 'some-other-import';

import { OccasionSelectType } from './types';

function a(arg1: Node) {}

type t2 = OtherNode;
type Comp1 = StatelessFunctionalComponent;
let Comp2: StatelessFunctionalComponent;

const t3 = React2;
const t4 = React;

class Test extends Component {}

const test = <Fragment>Some Content</Fragment>;
const test2 = <Frag key={'test'}>Some Content</Frag>;

export const Requests = () => {
  const [occasion, setOccasion] = useState<OccasionSelectType>(null);
  return null;
};

arg.reduce<GenericNode[]>(() => {}, {});
arg.reduce<{ a: { b: { c: GenericNode2 }[] } }>(() => {}, {});
"
`;
